Source/RapyutaSimulationPlugins/Private/Sensors/ROS2CameraComponent.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020-2021 Rapyuta Robotics Co., Ltd.
+
+#include "Sensors/ROS2CameraComponent.h"
+
+UROS2CameraComponent::UROS2CameraComponent()
+{
+    // component initialization
+    SceneCaptureComponent = CreateDefaultSubobject<USceneCaptureComponent2D>(TEXT("SceneCaptureComponent"));
+    RenderTarget = CreateDefaultSubobject<UTextureRenderTarget2D>(TEXT("RenderTarget"));
+
+    // Initialize capture and texture component
+    SceneCaptureComponent->SetupAttachment(this);
+}
+
+void UROS2CameraComponent::BeginPlay()
+{
+	Super::BeginPlay();	
+    Init();
+}
+
+void UROS2CameraComponent::Init()
+{
+    SceneCaptureComponent->FOVAngle = FieldOfView;
+    SceneCaptureComponent->OrthoWidth = OrthoWidth;
+    
+    RenderTarget->InitCustomFormat(Width, Height, EPixelFormat::PF_B8G8R8A8, true);
+    SceneCaptureComponent->TextureTarget = RenderTarget;
+
+    // Initialize image data
+    Data.width = Width;
+    Data.height = Height;
+    Data.encoding = Encoding;
+    Data.step = Width * 3; // todo should be variable based on encoding
+    Data.data.AddUninitialized(Width * Height * 3);
+
+    // Node and publisher initialize
+    FActorSpawnParameters SpawnParamsNode;
+    Node = GetWorld()->SpawnActor<AROS2Node>(AROS2Node::StaticClass(), SpawnParamsNode);
+    Node->Name = NodeName.IsEmpty() ? TEXT("UE4CameraNode_" + FGuid::NewGuid().ToString()) : NodeName;
+    Node->Namespace = FString();
+    Node->Init();
+
+    Publisher =  NewObject<UROS2Publisher>(this, UROS2Publisher::StaticClass());
+    Publisher->RegisterComponent();
+    Publisher->TopicName = TopicName;
+    Publisher->PublicationFrequencyHz = PublishFreq;
+    Publisher->MsgClass = UROS2ImageMsg::StaticClass();
+    
+    Publisher->UpdateDelegate.BindDynamic(this, &UROS2CameraComponent::MessageUpdate);
+    Node->AddPublisher(Publisher);
+    Publisher->Init(UROS2QoS::KeepLast);
+}
+
+void UROS2CameraComponent::MessageUpdate(UROS2GenericMsg *TopicMessage)
+{
+    UROS2ImageMsg *Message = Cast<UROS2ImageMsg>(TopicMessage);
+    Message->SetMsg(GetData());
+}

```

Comment:
We now have a template for ~ROS2 Publisher classes, like in [UROS2ClockPublisher](https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/blob/main/Source/RapyutaSimulationPlugins/Private/Tools/ROS2ClockPublisher.cpp), so should we have a class for `UROS2ImagePublisher` for consistency, also helping make this setup a bit shorter? then we only need to configure `TopicName` & `PublicationFrequencyHz` here.

Reply:
I think, it is too much for now since we have only this camera class which publish image. since getdata is tightly coupled with cameracomponent. we can do same as lidar to make publisher class have ROS2CameraComponent, 

we can revist this thing when we add other camera such as depth in the near feature.